riscv/triggers.cc

```diff
@@ -62,6 +65,46 @@ void trigger_t::tdata3_write(processor_t * const proc, const reg_t val) noexcept
     sselect = SSELECT_IGNORE;
 }
 
+bool trigger_t::textra_match(processor_t * const proc) const noexcept
+{
+  auto xlen = proc->get_xlen();
+  auto hsxlen = proc->get_xlen(); // use xlen since no hsxlen
+  state_t * const state = proc->get_state();
+
+  assert(sselect <= SSELECT_MAXVAL);
+  if (sselect == SSELECT_SCONTEXT) {
+    reg_t mask = (reg_t(1) << ((xlen == 32) ? CSR_TEXTRA32_SVALUE_LENGTH : CSR_TEXTRA64_SVALUE_LENGTH)) - 1;
+    assert(CSR_TEXTRA32_SBYTEMASK_LENGTH < CSR_TEXTRA64_SBYTEMASK_LENGTH);
+    for (int i = 0; i < CSR_TEXTRA64_SBYTEMASK_LENGTH; i++)
+      if (sbytemask & (1 << i))
+        mask &= 0xff << (i * 8);
+    if ((state->csrmap[CSR_SCONTEXT]->read() ^ svalue) & mask)
+      return false;
+  } else if (sselect == SSELECT_ASID) {
+    reg_t satp = state->satp->read();
+    reg_t satp_asid = xlen == 32 ? SATP32_ASID : SATP64_ASID;
+    reg_t asid = get_field(satp,  satp_asid);
+    if ((asid ^ svalue) & ((1 << ASIDMAX(xlen)) - 1))
+      return false;
+  }
+
+  reg_t mhvalue_length = xlen == 32 ? CSR_TEXTRA32_MHVALUE_LENGTH : CSR_TEXTRA64_MHVALUE_LENGTH;
+  if (mhselect == MHSELECT_MCONTEXT) {
+    if ((state->csrmap[CSR_MCONTEXT]->read() ^ mhvalue) & ((1 << mhvalue_length) - 1))
```

Comment:
With `scontext` and `mcontext` both being hardwired to 0, will this code ever do anything useful?